src/lib/archwizard.module.ts

```diff
@@ -15,6 +15,13 @@ import {WizardCompletionStepDirective} from './directives/wizard-completion-step
 import {WizardStepSymbolDirective} from './directives/wizard-step-symbol.directive';
 import {WizardStepTitleDirective} from './directives/wizard-step-title.directive';
 import {WizardStepDirective} from './directives/wizard-step.directive';
+import {NAVIGATION_MODE_FACTORY, NavigationModeFactory} from './navigation/navigation-mode-factory.interface';
+import {BaseNavigationModeFactory} from './navigation/navigation-mode-factory.provider';
+
+
+export interface ArchwizardModuleConfig {
+  navigationModeFactory?: new() => NavigationModeFactory;
```

Comment:
I never saw `new() => ...` before. Therefore, can you explain what it means? :)

Reply:
`() => ...` is a shorthand for `{(): ...}`. Both define a function taking no arguments and returning `...`

When `new` is added, it requires the function to be a constructor (which basically implies a class, in our case a class implementing `NavigationModeFactory`).

In other words, here we require a reference to a class implementing `NavigationModeFactory`.

Reply:
Why don't we use `navigationModeFactory: NavigationModeFactory`? Is there a benefit we achieve by using a constructor for `NavigationModeFactory` objects?

Reply:
Hmm, it did not cross my mind. I am so used to Angular dependency injection (`providers` etc.), which usually takes factories/constructors, that I did not notice that a plain instance would be more convenient in our case. I will fix it.
It is excellent that you noticed it!

Reply:
Fixed in 016b542